cdc/sink/simple_mysql_tester.go

```diff
@@ -0,0 +1,263 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sink
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"net/url"
+	"strings"
+	"sync"
+
+	dmysql "github.com/go-sql-driver/mysql"
+	"github.com/pingcap/errors"
+	"github.com/pingcap/failpoint"
+	"github.com/pingcap/log"
+	"github.com/pingcap/ticdc/cdc/model"
+	"github.com/pingcap/ticdc/pkg/config"
+	cerror "github.com/pingcap/ticdc/pkg/errors"
+	"github.com/pingcap/ticdc/pkg/filter"
+	"github.com/pingcap/ticdc/pkg/quotes"
+	"go.uber.org/zap"
+)
+
+func init() {
+	failpoint.Inject("SimpleMySQLSinkTester", func() {
+		sinkIniterMap["simple-mysql"] = func(ctx context.Context, changefeedID model.ChangeFeedID, sinkURI *url.URL,
+			filter *filter.Filter, config *config.ReplicaConfig, opts map[string]string, errCh chan error) (Sink, error) {
+			return newSimpleMySQLSink(ctx, sinkURI, config, opts)
+		}
+	})
+}
+
+type simpleMySQLSink struct {
+	enableOldValue      bool
+	enableCheckOldValue bool
+	db                  *sql.DB
+	rowsBuffer          []*model.RowChangedEvent
+	rowsBufferLock      sync.Mutex
+}
+
+func newSimpleMySQLSink(ctx context.Context, sinkURI *url.URL, config *config.ReplicaConfig, opts map[string]string) (*simpleMySQLSink, error) {
+	var db *sql.DB
+
+	// dsn format of the driver:
+	// [username[:password]@][protocol[(address)]]/dbname[?param1=value1&...&paramN=valueN]
+	username := sinkURI.User.Username()
+	password, _ := sinkURI.User.Password()
+	port := sinkURI.Port()
+	if username == "" {
+		username = "root"
+	}
+	if port == "" {
+		port = "4000"
+	}
+
+	dsnStr := fmt.Sprintf("%s:%s@tcp(%s:%s)/?multiStatements=true", username, password, sinkURI.Hostname(), port)
+	dsn, err := dmysql.ParseDSN(dsnStr)
+	if err != nil {
+		return nil, cerror.WrapError(cerror.ErrMySQLInvalidConfig, err)
+	}
+
+	// create test db used for parameter detection
+	if dsn.Params == nil {
+		dsn.Params = make(map[string]string, 1)
+	}
+	testDB, err := sql.Open("mysql", dsn.FormatDSN())
+	if err != nil {
+		return nil, errors.Annotate(
+			cerror.WrapError(cerror.ErrMySQLConnectionError, err), "fail to open MySQL connection when configuring sink")
+	}
+	defer testDB.Close()
+
+	db, err = sql.Open("mysql", dsnStr)
+	if err != nil {
+		return nil, errors.Annotate(
+			cerror.WrapError(cerror.ErrMySQLConnectionError, err), "Open database connection failed")
+	}
+	err = db.PingContext(ctx)
+	if err != nil {
+		return nil, errors.Annotate(
+			cerror.WrapError(cerror.ErrMySQLConnectionError, err), "fail to open MySQL connection")
+	}
+
+	sink := &simpleMySQLSink{
+		db:             db,
+		enableOldValue: config.EnableOldValue,
```

Comment:
The test case doesn't provide a changefeed config file, whether the `old-value` is enabled?

https://github.com/leoppro/ticdc/blob/9f85d2c4d2aa8bf00d488f4f3314b510e6acb1f5/integration/framework/mysql/single_table.go#L50